android5.0/Camera2Basic/Camera2Basic/Camera2BasicFragment.cs

```diff
@@ -4,524 +4,1050 @@
-				alert.SetMessage ("This device doesn't support Camera2 API.");
-				alert.SetPositiveButton (Android.Resource.String.Ok, new MyDialogOnClickListener (this));
-				return alert.Show();
-
-			}
-		}
-
-		private class MyDialogOnClickListener : Java.Lang.Object,IDialogInterfaceOnClickListener
-		{
-			ErrorDialog er;
-			public MyDialogOnClickListener(ErrorDialog e)
-			{
-				er = e;
-			}
-			public void OnClick(IDialogInterface dialogInterface, int i)
-			{
-				er.Activity.Finish ();
-			}
-		}
-	}
+    public class Camera2BasicFragment : Fragment, View.IOnClickListener, FragmentCompat.IOnRequestPermissionsResultCallback
+    {
+        private static readonly SparseIntArray ORIENTATIONS = new SparseIntArray();
+        private static readonly int REQUEST_CAMERA_PERMISSION = 1;
+        private static readonly string FRAGMENT_DIALOG = "dialog";
+
+        // Tag for the {@link Log}.
+        private static readonly string TAG = "Camera2BasicFragment";
+
+        // Camera state: Showing camera preview.
+        private const int STATE_PREVIEW = 0;
+
+        // Camera state: Waiting for the focus to be locked.
+        private const int STATE_WAITING_LOCK = 1;
+
+        // Camera state: Waiting for the exposure to be precapture state.
+        private const int STATE_WAITING_PRECAPTURE = 2;
+
+        //Camera state: Waiting for the exposure state to be something other than precapture.
+        private const int STATE_WAITING_NON_PRECAPTURE = 3;
+
+        // Camera state: Picture was taken.
+        private const int STATE_PICTURE_TAKEN = 4;
+
+        // Max preview width that is guaranteed by Camera2 API
+        private static readonly int MAX_PREVIEW_WIDTH = 1920;
+
+        // Max preview height that is guaranteed by Camera2 API
+        private static readonly int MAX_PREVIEW_HEIGHT = 1080;
+
+        // TextureView.ISurfaceTextureListener handles several lifecycle events on a TextureView
+        private Camera2BasicSurfaceTextureListener mSurfaceTextureListener;
+        private class Camera2BasicSurfaceTextureListener : Java.Lang.Object, TextureView.ISurfaceTextureListener
+        {
+            public Camera2BasicFragment owner;
+
+            public Camera2BasicSurfaceTextureListener(Camera2BasicFragment owner)
+            {
+                this.owner = owner;
+            }
+
+            public void OnSurfaceTextureAvailable(Android.Graphics.SurfaceTexture surface, int width, int height)
+            {
+                owner.OpenCamera(width, height);
+            }
+
+            public bool OnSurfaceTextureDestroyed(Android.Graphics.SurfaceTexture surface)
+            {
+                return true;
+            }
+
+            public void OnSurfaceTextureSizeChanged(Android.Graphics.SurfaceTexture surface, int width, int height)
+            {
+                owner.ConfigureTransform(width, height);
+            }
+
+            public void OnSurfaceTextureUpdated(Android.Graphics.SurfaceTexture surface)
```

Comment:
It belongs to interface `TextureView.ISurfaceTextureListener`. So, it forces you to implement something in.